modules/keymaps.nix

```diff
@@ -52,6 +51,36 @@ with lib;
       };
     in
     {
+      # Deprecate `lua` keymap option
+      # TODO upgrade to an assertion (removal notice) in 24.11
+      # TODO remove entirely in 25.05?
```

Comment:
Not sure what the best deprecation strategy is, open to suggestions and alternative TODO wording.

I'm working on the assumption we'll want to give users upgrading from one stable nixos version to another at least one chance to see a deprecation warning without their config breaking.

Reply:
If we merge this right now, it will get into 24.05.
So, we should be able to remove it definitely in 24.11.

Reply:
We can even backport the "assertion" in 24.05 in a few months.

Reply:
> So, we should be able to remove it definitely in 24.11.

When we remove the option, should we keep a custom assertion message for a while or just rely on the module system's "no such option" assertion?

> We can even backport the "assertion" in 24.05 in a few months.

Should I add some datestamps to the TODO comments?